opcua/common/node.py

```diff
@@ -404,15 +404,9 @@ def read_event_history(self, starttime=None, endtime=None, numvalues=0, evtypes=
         if not type(evtypes) in (list, tuple):
             evtypes = [evtypes]
 
-        # FIXME not a very nice way to make sure events.get_filter gets a list of nodes...
-        evtype_nodes = []
-        for evtype in evtypes:
-            if not isinstance(evtype, Node):
-                evtype_nodes.append(Node(self.server, ua.NodeId(evtype)))  # make sure we have a list of Node objects
-            else:
-                evtype_nodes.append(evtype)
+        evtypes = [Node(self.server, evtype) for evtype in evtypes]
 
```

Comment:
Should work. I changed init of Node a while back to accept crazy things. If you think a test I missing feel free to add it 😃.
I see you still have a comment about things to check in code. Will have a look.